src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -693,7 +693,7 @@ impl<'a, 'tcx> BorrowckCtxt<'a, 'tcx> {
             .span_err(span,
                       (format!("partial reinitialization of uninitialized \
                                structure `{}`",
-                               self.loan_path_to_string(lp))).as_slice());
+                               self.loan_path_to_string(lp))).as_str());
```

Comment:
I missed this one, although further poking seems to suggest that the &'s are unnecesary to get the String returned by format! to coerce